logbook-api/src/main/java/org/zalando/logbook/BaseHttpMessage.java

```diff
@@ -35,22 +35,42 @@
 
     Origin getOrigin();
 
-    ListMultimap<String, String> getHeaders();
+    Map<String, List<String>> getHeaders();
 
     String getContentType();
 
     Charset getCharset();
 
-    class Headers {
+    class HeadersBuilder {
+
+        private final Map<String, List<String>> headers;
 
-        Headers() {
+        public HeadersBuilder() {
```

Comment:
I am thinking of implementing Header and Headers implements Iterable<Header> classes instead of map of lists as this will allow to extend it in future.